src/test/ui/underscore-lifetime/underscore-lifetime-binders.stderr

```diff
@@ -30,7 +30,7 @@ error[E0106]: missing lifetime specifier
 LL | fn foo2(_: &'_ u8, y: &'_ u8) -> &'_ u8 { y }
    |                                   ^^ expected lifetime parameter
    |
-   = help: this function's return type contains a borrowed value, but the signature does not say whether it is borrowed from `_` or `y`
+   = help: this function's return type contains a borrowed value, but the signature does not say whether it is borrowed from argument 1 or `y`
```

Comment:
@estebank Just to check with you, this is an improvement, right?

Reply:
@eddyb It'd might be even better if we added spans to it, like below, but this is an improvement in my eyes

```
error[E0106]: missing lifetime specifier
  --> $DIR/underscore-lifetime-binders.rs:16:35
   |
LL | fn foo2(_: &'_ u8, y: &'_ u8) -> &'_ u8 { y }
   |                                   ^^ expected lifetime parameter
   |
help: this function's return type contains a borrowed value, but the signature does not say whether it is borrowed from argument 1 or `y`
   |
LL | fn foo2(_: &'_ u8, y: &'_ u8) -> &'_ u8 { y }
   |         ^          ^

```

Reply:
It would be even better if it pointed at the lifetimes in the signature (which is information we could collect).

Reply:
I think it might be best if we left this as a follow-up (it's a bit out of scope for this PR). I'd be happy to do it after this though.